### PR TITLE
Added support for Travis-CI

### PR DESCRIPTION
**Pull request by kajdreef**
Added support for Travis-CI at the moment the building process is relative slow (±2 minutes), so I think I should add caching to it, to cache Tacoco and the Hamcrest dependencies. However, I wanted to open a discussion about what other checks should be placed in the integration process (style checkers for example)?

### CODE REVIEW

**Comment by VijayKrishna** [comment]
@kajdreef In terms of additional integration checks as part of the CI, I would be interested in the results of your use of various static checkers on Blinky. Specifically, I would look at answering the following questions before adding any such checker:

1. What use do we get out of using a static analyzer?
2. How are the results of a static analyzer actionable?
3. How easy would it be to debug a build failure using only the logs of the checker?

Can you list a bunch of style checkers or other integration checks that you have in mind? We can then run through them real quick and figure how useful they are overall. :)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+- cd primitive-hamcrest  
+- mvn install
+- cd ..
+- git clone https://github.com/spideruci/tacoco.git
+- cd tacoco
+- mvn install
+- cd ..
+
+after_success:
+- mvn clean test -P full
```
Comment (VijayKrishna) [inline .travis.yml]: what happens when you do a `mvn clean verify -P full`?

**Comment by VijayKrishna** [comment]
Here is one static check that can bring clear value to Blinky: Import Control http://checkstyle.sourceforge.net/config_imports.html#ImportControl. It would be nice to ensure that we never get into a state of dependency hell :)

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+language: java
+
+jdk: 
+- oraclejdk8
+
+before_install:
```
Comment (VijayKrishna) [inline .travis.yml]: Can we move all these instructions to a shell script called `before_install.sh` that lives with this project?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+language: java
+
+jdk: 
+- oraclejdk8
```
Comment (VijayKrishna) [inline .travis.yml]: we work with Java8? :)

File: .travis.yml (hunk 1 above)
Comment (kajdreef) [reply .travis.yml]: That should work haven't tried it out yet. What is the difference between the verify and test goal? I see it also runs blinky-diagnostics, but does it also run all the unit tests? I quickly went through it manually and I think it does right? So shall I change test to verify?

File: .travis.yml (hunk 2 above)
Comment (kajdreef) [reply .travis.yml]: That is a good idea! I will make sure I will do that!

File: .travis.yml (hunk 3 above)
Comment (kajdreef) [reply .travis.yml]: Yes, I have setup travis-ci for now on my [Fork](https://travis-ci.org/kajdreef/blinky) to test it all out. When compiling locally java9 doesn't work, but I haven't looked into that yet. :)

**Pull request merged**